server/src/main/java/fr/uga/l3miage/tp1/exo3/models/AssociationEntity.java

```diff
@@ -0,0 +1,30 @@
+package fr.uga.l3miage.tp1.exo3.models;
+
+import fr.uga.l3miage.tp1.exo3.enums.AssociationType;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "association")
+public class AssociationEntity {
+    @Id
+    @Column(name = "name")
+    private String name ;
+
+    @Column(name = "president_name", nullable = false)
+    private String presidentName ;
+
+    @Column(name = "treasurer_name", nullable = false)
+    private String treasurerName ;
+
+    @Column(name = "secretary_name")
+    private String secretaryName ;
+
+    @Column(name = "association_type")
+    @Enumerated(EnumType.STRING)
+    private AssociationType associationType ;
+
+    @ManyToOne
+    @JoinColumn(name = "federation_name")
+    private FederationEntity federation ;
+}
```

Comment:
idem pour les @Column
```suggestion
@Entity
@Table(name = "association")
public class AssociationEntity {
    @Id
    private String name ;

    @Column(nullable = false)
    private String presidentName ;

    @Column(nullable = false)
    private String treasurerName ;

    private String secretaryName ;

    @Enumerated(EnumType.STRING)
    private AssociationType associationType ;

    @ManyToOne
    private FederationEntity federation ;
}
```
et idem je `@JoinColumn` n'est pas utile ici, c'est le rôle du `mappedby`